Signal/src/ViewControllers/Utils/MessageRecipientStatusUtils.swift

```diff
@@ -153,67 +154,51 @@ class MessageRecipientStatusUtils: NSObject {
+                                         comment: "message footer while attachment is uploading")
             } else {
                 return NSLocalizedString("MESSAGE_STATUS_SENDING",
-                                         comment:"message status while message is sending.")
+                                         comment: "message status while message is sending.")
             }
-        case .sentToService:
-            let recipientReadMap = outgoingMessage.recipientReadMap
-            if recipientReadMap.count > 0 {
-                return NSLocalizedString("MESSAGE_STATUS_READ", comment:"message footer for read messages")
+        case .sent:
+            if outgoingMessage.readRecipientIds().count > 0 {
+                return NSLocalizedString("MESSAGE_STATUS_READ", comment: "message footer for read messages")
             }
-
-            let recipientDeliveryMap = outgoingMessage.recipientDeliveryMap
-            if recipientDeliveryMap.count > 0 {
+            if outgoingMessage.deliveredRecipientIds().count > 0 {
                 return NSLocalizedString("MESSAGE_STATUS_DELIVERED",
-                                         comment:"message status for message delivered to their recipient.")
+                                         comment: "message status for message delivered to their recipient.")
             }
-
-            if outgoingMessage.wasDelivered {
-                return NSLocalizedString("MESSAGE_STATUS_DELIVERED",
-                                         comment:"message status for message delivered to their recipient.")
-            }
-
             return NSLocalizedString("MESSAGE_STATUS_SENT",
-                                     comment:"message footer for sent messages")
+                                     comment: "message footer for sent messages")
         default:
             owsFail("Message has unexpected status: \(outgoingMessage.messageState).")
             return NSLocalizedString("MESSAGE_STATUS_SENT",
-                                     comment:"message footer for sent messages")
+                                     comment: "message footer for sent messages")
         }
     }
 
     // This method is per-message and "biased towards failure".
     // See comments above.
     public class func recipientStatus(outgoingMessage: TSOutgoingMessage) -> MessageRecipientStatus {
         switch outgoingMessage.messageState {
-        case .unsent:
+        case .failed:
             return .failed
-        case .attemptingOut:
+        case .sending:
             if outgoingMessage.hasAttachments() {
                 return .uploading
             } else {
                 return .sending
             }
-        case .sentToService:
-            let recipientReadMap = outgoingMessage.recipientReadMap
-            if recipientReadMap.count > 0 {
+        case .sent:
+            if outgoingMessage.readRecipientIds().count > 0 {
```

Comment:
I've DRY'd up a little, but we still need to similar looking logic in two places since we're doing two separate mappings:

* Per-message status `TSOutgoingMessageState` -> renderable receipt state `MessageReceiptStatus`.
* Per-Recipient status `OWSOutgoingMessageRecipientState` -> renderable receipt state `MessageReceiptStatus`.